app/components/new-file/new-file.component.ts

```diff
@@ -21,58 +22,46 @@ export class NewFileResult {
 })
 export class NewFileComponent implements ModalComponent, AfterViewInit {
     set modalInfo(modalInfo: ModalInfo) {
-        let plugins: string[][] = modalInfo.data;
-        let availablePlugins: any = { subLists: {} };
+        const plugins: PluginData[] = modalInfo.data;
+        // This code is extremely evil but necessary.
```

Comment:
The reason it is necessary is because they need toString to work properly with the collapsible list component. However, prototype information gets stripped out when something is serialized so I just add it back in. It's probably still unsafe, though it is safer than it first appears.